src/gui/properties/peerlistwidget.cpp

```diff
@@ -87,15 +88,20 @@ PeerListWidget::PeerListWidget(PropertiesWidget *parent)
     m_proxyModel->setSourceModel(m_listModel);
     m_proxyModel->setSortCaseSensitivity(Qt::CaseInsensitive);
     setModel(m_proxyModel);
-    //Explicitly set the column visibility. When columns are added/removed
-    //between versions this prevents some of them being hidden due to
-    //incorrect restoreState() being used.
-    for (unsigned int i = 0; i < PeerListDelegate::IP_HIDDEN; i++)
-        showColumn(i);
     hideColumn(PeerListDelegate::IP_HIDDEN);
     hideColumn(PeerListDelegate::COL_COUNT);
     if (!Preferences::instance()->resolvePeerCountries())
         hideColumn(PeerListDelegate::COUNTRY);
+    //Ensure that at least one column is visible at all times
+    bool atLeastOne = false;
+    for (unsigned int i = 0; i < PeerListDelegate::COL_COUNT; i++) {
```

Comment:
You should use `for (unsigned int i = 0; i < PeerListDelegate::IP_HIDDEN; i++) {` instead.
`IP_HIDDEN` should always be hidden.